pkg/kubectl/cmd/util/factory.go

```diff
@@ -251,8 +251,11 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			return generator, ok
 		},
 		CanBeExposed: func(kind string) error {
-			if kind != "ReplicationController" && kind != "Service" && kind != "Pod" {
-				return fmt.Errorf("invalid resource provided: %v, only a replication controller, service or pod is accepted", kind)
+			switch kind {
+			case "ReplicationController", "Service", "Pod":
+				// nothing to do here
+			default:
+				return fmt.Errorf("cannot expose a %s", kind)
```

Comment:
This is actually one of the reasons behind this change. We don't follow this pattern elsewhere in the factory (hardcoding supported types in error messages) and this is already known because it exists in the help string: https://github.com/kubernetes/kubernetes/blob/e05819f36a2fffa03a420ac3a246e0bedf18a487/pkg/kubectl/cmd/expose.go#L38